paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -85,55 +92,74 @@ public Path changelogDirectory() {
         return new Path(tablePath + "/changelog");
     }
 
-    public Path longLivedChangelogPath(long snapshotId) {
-        return new Path(tablePath + "/changelog/" + CHANGELOG_PREFIX + snapshotId);
+    public Path changelogDirectory(String branchName) {
```

Comment:
What is the design? Why do need to pass branch here when there is already a branch in the constructor?

Reply:
> What is the design? Why do need to pass branch here when there is already a branch in the constructor?

I have mostly opened up interfaces for shoelace branch parameters and those that do not carry branch parameters. The interfaces with branches are mainly provided for ‘BranchManager’ and can also be used for testing

Reply:
I have mostly opened up methods with and without branch parameters. The methods with branch parameters are mainly for BranchManager and test preparation

Reply:
So, please modify `BranchManager` to use separate `SnapshotManager`s instead of using this method.